supervised/preprocessing/eda.py

```diff
@@ -162,3 +162,161 @@ def compute(X, y, eda_path):
             fout.close()
         except Exception as e:
             logger.error(f"There was an issue when running EDA. {str(e)}")
+
+    @staticmethod
+    def extensive_eda(df, y, save_path):
+
+        X = df.copy(deep=True)
+        # Check for empty dataframes in params
+        if not isinstance(X, pd.DataFrame):
+            raise ValueError("X should be a dataframe")
+        if X.shape[0] != len(y):
+            raise ValueError("X and y should have same number of samples")
+
+        if X.shape[1] > MAXCOL:
+            X = X.iloc[:, :MAXCOL]
+            warnings.warn(
+                f"AutoML EDA column limit exceeded! running for first {MAXCOL} columns"
+            )
+
+        ## replace forbidden file characters with '_'  symbol
+        X.columns = [re.sub(r'[\\/*?:"<>|]', "_", x) for x in X.columns]
+
+        if save_path:
+            if not os.path.exists(save_path):
+                os.mkdir(save_path)
+        else:
+            raise ValueError("provide a valid path to save plots")
```

Comment:
Maybe change the exception msg: "Please provide a valid path to save the Extensive EDA results"?